integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java

```diff
@@ -179,7 +185,13 @@ private JAXRSServerFactoryBean createFactoryInstance(final Application applicati
                 classified.get(Path.class).add(singleton);
             }
         }
-        
+
+        // now loop through the classes
+        Set<Class<?>> classes = application.getClasses();
+        if (!classes.isEmpty()) {
+            classified.get(Path.class).addAll(loadServices(beanManager, classes));
+            classified.get(Provider.class).addAll(loadProviders(beanManager, classes));
```

Comment:
Shoot, thanks.  I actually wrote the code for features, but then removed it in my formatting clean up.